python-api/tests/integrationtests/baseclass.py

```diff
@@ -0,0 +1,412 @@
+        self.log.debug("saving to %s", filename)
+        scipy.misc.imsave(
+            filename,
+            diff.astype(numpy.uint8))
+
+        if self.is_running_in_ci():
+            self.log.warn("uploaded difference frame")
+            self.upload_image(filename)
+        else:
+            self.log.warn("saved difference frame as: %s",
+                          filename)
+
+    def upload_image(self, filename):
+        """ upload the specified image to imgurl"""
+        self.log.debug("calling imgurbash.sh %s", filename)
+        subprocess.call([
+            'bash',
+            './imgurbash.sh',
+            filename
+        ], timeout=5)
+
+
+class IntegrationTestbaseVideo(IntegrationTestbaseCompare):
+    """ Testbase used to test video-pipeline results
+    """
+
+    PORT_RED = 3003
+    PORT_GREEN = 3004
+    PORT_BLUE = 3005
+    PORTS = (PORT_RED, PORT_GREEN, PORT_BLUE, )
+
+    def expect_video_frame(self, filename, port=3001, timeout=5):
+        """Read frames from the server and compare them against filename.
+        Return when a match is found or timeout seconds have passed
+        Source-Port defaults to 3001=video compose-port"""
+
+        self.expect_frame(
+            filename, port,
+            turns=timeout,
+            timeout=1)
+
+    def get_capture_command(self, port, filename_pattern):
+        """Construct a Gstreamer-Pipeline which receives frames,
+        converts them to PNG-Files and writed them to Files.
+        A gst-alunch-command is returned"""
+
+        return ("gst-launch-1.0 "
+                "tcpclientsrc host=localhost port=%u ! "
+                "gdpdepay ! "
+                "videoconvert ! "
+                "pngenc ! "
+                "multifilesink location=%s") % (port, filename_pattern)
+
+    def setup_test(self):
+        """Setup Server, Controller and two Video-Test-Sources"""
+        self.setup_server()
+        self.setup_controller()
+
+        self.log.info("starting up 1st VideoSrc")
+        self.new_test_video(pattern=VideoSrc.PATTERN_RED)
+
+        self.log.info("starting up 2nd VideoSrc")
+        self.new_test_video(pattern=VideoSrc.PATTERN_GREEN)
+
+        self.log.info("starting up 3rd VideoSrc")
+        self.new_test_video(pattern=VideoSrc.PATTERN_BLUE)
+
+        self.serv.wait_for_output('tcpserversink name=sink', count=3)
+
+        self.log.info("waiting for end of initial transition")
+        self.serv.wait_for_output('ending transition')
+
+
+class IntegrationTestbaseAudio(IntegrationTestbaseCompare):
+    """ Testbase used to test video-pipeline results
+    """
+
+    PORT_110 = 3003
+    # PORT_220 = 3004
+    PORTS = (PORT_110, )
+
+    def expect_audio_spectrascope(self, filename, port=3003, timeout=5):
+        """Read frames from the server and compare them against filename.
+        Return when a match is found or timeout seconds have passed
+        Source-Port defaults to 3001=video compose-port"""
```

Comment:
fixed the comment, restructuring the doc-comments should be another PR based on #210